packages/expo-updates/android/src/main/java/expo/modules/updates/db/dao/AssetDao.kt

```diff
@@ -89,10 +89,21 @@ abstract class AssetDao {
 
   fun mergeAndUpdateAsset(existingEntity: AssetEntity, newEntity: AssetEntity) {
     // if the existing entry came from an embedded manifest, it may not have a URL in the database
-    if (newEntity.url != null && existingEntity.url == null) {
+    var shouldUpdate = false
+    if ((newEntity.url != null && existingEntity.url == null) || newEntity.url != existingEntity.url) {
       existingEntity.url = newEntity.url
+      shouldUpdate = true
+    }
+
+    if (newEntity.extraRequestHeaders?.equals(existingEntity.extraRequestHeaders) == true) {
```

Comment:
I think this is incorrect. It should update the headers in the existing entity if:
- headers are not equal
- headers in existing entity are null and new ones aren't